router.go

```diff
@@ -61,7 +62,13 @@ type Route struct {
 	Handlers []Handler `json:"-"`      // Ctx handlers
 }
 
-func (r *Route) match(detectionPath, path string, params *[maxParams]string) (match bool) {
+func (r *Route) match(detectionPath, path string, params *[maxParams]string, matcher ...utils.RegexMatch) (match bool) {
```

Comment:
don't like having a functionality with too many parameters 

cleancode says that this should be rethought and reimplemented

Reply:
may be we can add `defaultMatcher` variable (local) and set it inside of the `New()` instead of adding new param.